package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +190,73 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+M.PipePolicies = {
+	DISCARD=-1,
+	INHERIT=0,
+	CREATE=1
+}
+
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.popen3(policies, path, ...)
```

Comment:
Having the policies first feels a bit weird, especially as a user might want to the policies (if all FDs are to be inherited). I suggest making the command a table instead, so the signature would become
```lua
function popen(cmd, fds)
```

Reply:
Where `cmd` contains both `path` and the command table (`argt`)?

This change would drop the similarity to `execp`s signature.
https://luaposix.github.io/luaposix/modules/posix.unistd.html#execp

Reply:
I like `execp`'s signature, let's do it like that.

Reply:
The problem is then the trailing vararg; which leaves us with the options of
`path, policies, ...`
and the current
`policies, path, ...`.

Which do you favor? Or did you intend something else with 
> let's do it like that

?

Reply:
I don't think the function should be vararg, `execp` isn't vararg either. `popen(path, argt, options)` would make sense to me.

Reply:
resolved in a66e8b815dc1ba40e5e99efe07d7e2bb264001d2 and e9e42fa261a9089bf67a1d085517189a62eebc3e